src/middleware/isnt-webhook.ts

```diff
@@ -4,14 +4,14 @@ import { IncomingMessage } from "http"
 
 export function isntWebhook(request: IncomingMessage, options: { path?: string }) {
   // GitHub sends all events as POST requests
-  if (request.method !== "POST") {
+  if (request.method !== "POST" || !request.url) {
     return true;
   }
 
   // We must match the configured path to allow custom POST routes which include
   // the webhook route. For example if the webhook route is / then it would be
   // impossible to define a `POST /my/custom/app` route as the `POST /`.
-  if (request.url?.split("?")[0] !== options.path) {
+  if (request.url.split("?")[0] !== options.path) {
```

Comment:
If the `?` makes problems, we could do 

```suggestion
  if (typeof request.url !== 'string' || request.url.split("?")[0] !== options.path) {
```